src/targets/gpu/fuse_mlir.cpp

```diff
@@ -104,25 +104,42 @@ struct mlir_op
             MIGRAPHX_THROW("should have at least two inputs.");
 
         auto type = mod->get_output_shapes().front().type();
-        std::unordered_map<instruction_ref, shape> ins_shapes;
+        auto mod_params = mod->get_parameter_names();
+        std::sort(mod_params.begin(), mod_params.end());
+        std::unordered_map<instruction_ref, shape> mod_ins_shapes;
+        std::unordered_map<std::string, shape> adjusted_mod_param_shapes;
+        std::transform(inputs.begin(),
+                       inputs.end(),
+                       mod_params.begin(),
+                       std::inserter(adjusted_mod_param_shapes, adjusted_mod_param_shapes.end()),
+                       [](auto ps, auto name) { return std::make_pair(name, ps); });
         for(auto ins : iterator_for(*mod))
         {
-            if(ins->name() == "@literal" or ins->name() == "@param")
+            if(ins->name() == "@param")
             {
-                ins_shapes[ins] = ins->get_shape();
-                continue;
+                mod_ins_shapes[ins] =
+                    adjusted_mod_param_shapes[any_cast<builtin::param>(ins->get_operator())
+                                                  .parameter];
+                assert(ins->get_shape().type() == mod_ins_shapes[ins].type());
```

Comment:
This should be a throw or hard error. Also we should probably check the `.lens()` match as well.

Reply:
Done.